### ISSUE TITLE
Zip slip vulnerability

### ISSUE THREAD

**Issue by xcorail** [PowerShell/PowerShell#7985]
Hi there

[LGTM](https://lgtm.com) discovered a potential occurrence of the Zip Slip vulnerability. The alert is available [here](https://lgtm.com/projects/g/PowerShell/PowerShell/snapshot/af8e6c33674909b8f87fee88783d99dd0739946e/files/src/System.Management.Automation/help/UpdatableHelpSystem.cs?sort=name&dir=ASC&mode=heatmap&showExcluded=true#V1261)

For more details about this vulnerability : https://lgtm.com/rules/1506511188430/
Best regards

**Comment by iSazonov** [comment]
I don't see a problem - we use recommended `Path.Combine(destination, entry.FullName);`

**Comment by xcorail** [comment]
Hi @iSazonov 
```
The recommended way of writing an output file from a zip archive entry is to:

1. Use Path.Combine(destinationDirectory, archiveEntry.FullName) to determine the raw output path.
2. Use Path.GetFullPath(..) on the raw output path to resolve any directory traversal elements.
3. Use Path.GetFullPath(destinationDirectory + Path.DirectorySeparatorChar) to determine the fully resolved path of the destination directory.
4. Validate that the resolved output path StartsWith the resolved destination directory, aborting if this is not true.
```
The recommendation is to do 1, 2, 3 and 4, not only 1. `Path.Combine` does not perform any sanitization of the entry full path 

Cheers

**Comment by iSazonov** [comment]
@xcorail Thanks for clarify!

@TravisEz13 We have already this in CoreFX https://source.dot.net/#System.IO.Compression.ZipFile/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs,104

**Comment by TravisEz13** [comment]
Fixed in #8252
